src/compute/src/logging/timely.rs

```diff
@@ -183,14 +185,27 @@ pub(super) fn construct<A: Allocate>(
                     Datum::UInt64(u64::cast_from(target_port)),
                 ])
             });
+
+        let mut packer = PermutedRowPacker::new(TimelyLog::Addresses.into());
         let addresses = addresses
             .as_collection()
             .mz_arrange_core::<_, RowSpine<_, _, _, _>>(
                 Exchange::new(move |_| u64::cast_from(worker_id)),
                 "PreArrange Timely addresses",
                 compute_state.enable_arrangement_size_logging,
             )
-            .as_collection(move |id, address| create_address_row(*id, address, worker_id));
+            .as_collection({
+                move |id, address| {
+                    packer.pack_by_index(|packer, index| match index {
+                        0 => packer.push(Datum::UInt64(u64::cast_from(*id))),
+                        1 => packer.push(Datum::UInt64(u64::cast_from(worker_id))),
+                        2 => packer
+                            .push_list(address.iter().map(|i| Datum::UInt64(u64::cast_from(*i)))),
+                        _ => unreachable!(),
```

Comment:
Since this branch is not obviously unreachable by looking at the local context, could you add a helpful panic message?